File: my-folder/0141-linked-list-cycle/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
      set<ListNode*> s;
      while(head){
        if(s.contains(head)) return true;
        s.insert(head);
        head = head->next;
      }
      return false;  
    }
};

File: my-folder/0141-linked-list-cycle/solution.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function hasCycle(head: ListNode | null): boolean {
    if(!head) return false;
    let currPtr = head;
    let nextPtr = head.next;
    while(nextPtr){
        if(currPtr === nextPtr) return true;
        currPtr = currPtr.next;
        nextPtr = nextPtr.next != null ? nextPtr.next.next : null ;
    }

    return false;
};
